x-pack/plugins/lens/public/editor_frame_service/error_helper.ts

```diff
@@ -26,22 +27,37 @@ const isRequestError = (e: Error | RequestError): e is RequestError => {
   return false;
 };
 
+const isScriptedFieldError = (e: ElasticsearchErrorClause) => {
+  if ('failed_shards' in e) {
+    return e.failed_shards?.some(({ reason }) => reason?.type === 'script_exception');
+  }
+  return false;
+};
+
 interface ESError extends Error {
-  attributes?: { caused_by?: ElasticsearchErrorClause };
+  attributes?: ElasticsearchErrorClause;
 }
 
 const isEsError = (e: Error | ESError): e is ESError => {
   if ('attributes' in e) {
-    return e.attributes?.caused_by?.caused_by !== undefined;
+    return e.attributes?.caused_by !== undefined;
   }
   return false;
 };
 
-function getNestedErrorClause({
-  type,
-  reason,
-  caused_by: causedBy,
-}: ElasticsearchErrorClause): { type: string; reason: string } {
+function getNestedErrorClause(
+  e: ElasticsearchErrorClause
+): { type: string; reason: string; nestedError?: { type: string; reason: string } } {
+  if (isScriptedFieldError(e)) {
+    const nestedError = e.failed_shards![0].reason;
```

Comment:
I think you can't just look at the first shard for the script error. Here is a contrived example that I haven't tested:

* You have an index pattern with mapping conflicts, maybe one field is a keyword and one is a full text field. Full text fields aren't scriptable in aggs.
* Your script will fail on the indices that have the full text, but not the indices with keywords.

Reply:
I think this is partially true. I'll try to collect them all, but also in the `data` plugin all errors but the first one are discarded.